wgengine/monitor/polling.go

```diff
@@ -42,6 +44,21 @@ func (pm *pollingMon) Close() error {
 	return nil
 }
 
+func (pm *pollingMon) isCloudRun() bool {
+	// https: //cloud.google.com/run/docs/reference/container-contract#env-vars
+	if os.Getenv("K_REVISION") == "" || os.Getenv("K_CONFIGURATION") == "" ||
+		os.Getenv("K_SERVICE") == "" || os.Getenv("PORT") == "" {
+		return false
+	}
+	vers, err := os.ReadFile("/proc/version")
+	if err != nil {
+		pm.logf("Failed to read /proc/version: %q", err)
+		return false
+	}
+	v := strings.TrimSuffix(string(vers), "\n")
+	return v == "Linux version 4.4.0 #1 SMP Sun Jan 10 15:06:54 PST 2016"
```

Comment:
minor, but if you do this instead, it doesn't allocate:

```go
   return string(bytes.TrimSpace(vers)) == "Linux version 4.4.0 #1 SMP Sun Jan 10 15:06:54 PST 2016"
```

That is, `string([]byte) == string` doesn't allocate. And neither does `bytes.TrimSpace`.

Reply:
Done.